hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java

```diff
@@ -900,5 +902,19 @@ public static Map<String, List<ColumnFamilyHandle>> getCfHandleMap() {
     return dbNameToCfHandleMap;
   }
 
-
+  @Override
+  protected void finalize() throws Throwable {
+    if (!isClosed()) {
+      String warning = "RocksDatabase is not closed properly.";
+      if (LOG.isDebugEnabled()) {
+        String stackTrace =
+            Arrays.toString(Thread.currentThread().getStackTrace());
```

Comment:
I guess the stacktrace should be in the place where it got initialized

Reply:
This will have the GC thread stacktrace, it wouldn't be of much help here. I guess this can go along with the warning & need not go only on debug.

Reply:
You are right. Fixed it.